ADO.NET.DataLayer/Repositories/BaseRepository.cs

```diff
@@ -0,0 +1,157 @@
+                {
+                    var parameter = command.CreateParameter();
+                    parameter.ParameterName = "@" + param.Key;
+                    parameter.Value = param.Value;
+                    command.Parameters.Add(parameter);
+                }
+
+                OpenConnection();
+                command.ExecuteNonQuery();
+            }
+        }
+
+        public List<T> ConvertExecutedProcedureWithParamsToModel(string procedureName, Dictionary<string, object> parameters)
+        {
+            using (var command = _connection.CreateCommand())
+            {
+                command.CommandText = procedureName;
+                command.CommandType = CommandType.StoredProcedure;
+
+                foreach (var param in parameters)
+                {
+                    var parameter = command.CreateParameter();
+                    parameter.ParameterName = "@" + param.Key;
+                    parameter.Value = param.Value;
+                    command.Parameters.Add(parameter);
+                }
+
+                var readStream = command.ExecuteReader();
+
+                var result = new List<T>();
+                while (readStream.Read())
+                {
+                    var item = new T();
+                    foreach (var prop in item.GetType().GetProperties())
+                    {
+                        prop.SetValue(item, readStream[prop.Name]);
+                    }
+                    result.Add(item);
+                }
+
+                return result;
+            }
+        }
+
+        public List<T> ConvertConnectedQueryResultToModel(string query, Dictionary<string, object> parameters)
+        {
+            OpenConnection();
+
+            using (var command = _connection.CreateCommand())
+            {
+                command.CommandText = query;
+
+                foreach (var param in parameters)
+                {
+                    var parameter = command.CreateParameter();
+                    parameter.ParameterName = "@" + param.Key;
+                    parameter.Value = param.Value;
+                    command.Parameters.Add(parameter);
+                }
+
+                var readStream = command.ExecuteReader();
+
+                var result = new List<T>();
+                while (readStream.Read())
+                {
+                    var item = new T();
+                    foreach (var prop in item.GetType().GetProperties())
+                    {
+                        prop.SetValue(item, readStream[prop.Name]);
+                    }
+                    result.Add(item);
+                }
+
+                return result;
+            }
+        }
+
+        public DataTable ExecuteDisconnectedQuery(string query, Dictionary<string, object> parameters)
+        {
+            using (var connection = _connection)
+            {
+                OpenConnection();
+
+                using (var command = connection.CreateCommand())
```

Comment:
And here not need in using statement
https://learn.microsoft.com/en-us/dotnet/framework/data/adonet/executing-a-command#example